### ISSUE TITLE
Whitespace rule `check-decl` incorrectly flags definite assignment assertion

### ISSUE THREAD

**Issue by ericbf** [palantir/tslint#4448]
### Bug Report

- __TSLint version__: 5.12.0
- __TypeScript version__: 3.1.6
- __Running TSLint via__: Atom

#### TypeScript code being linted

```ts
class Dog {
	name!: string
}
```

with `tslint.json` configuration:

```json
{
	"rules": {
		"whitespace": [
			true,
			"check-decl"
		]
	}
}
```

#### Actual behavior

The definite assignment assertion is underlined with the error `missing whitespace`.

#### Expected behavior

No warning should be given, as this is not part of a "variable declaration"' (["check-decl" checks that variable declarations have whitespace around the equals token.](https://palantir.github.io/tslint/rules/whitespace/)) . If anything, this should be captured under the `typedef-whitespace` rule, with no interaction with the `whitespace` rule.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
